src/main/scala/monix/nio/tcp/AsyncSocketChannel.scala

```diff
@@ -23,103 +23,354 @@ import java.nio.channels.spi.AsynchronousChannelProvider
 import java.nio.channels.{ AsynchronousChannelGroup, AsynchronousSocketChannel, CompletionHandler }
 import java.util.concurrent.TimeUnit
 
-import monix.eval.Callback
+import monix.eval.{ Callback, Task }
 import monix.execution.Scheduler
 import monix.nio.internal.ExecutorServiceWrapper
 
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Future, Promise }
 import scala.util.control.NonFatal
 
+/**
```

Comment:
I appreciate the comments very much, but lets do it ScalaDoc style.

```scala
/** An asynchronous channel for reading, writing, and manipulating a TCP socket.
  *  
  *  On the JVM this is a wrapper around ...
  *  ...
  */
abstract class AsyncSocketChannel extends AutoCloseable
```

To configure IntelliJ IDEA, go to Settings -> Editor -> Code Style -> Scala -> ScadaDoc -> Check "Use ScalaDoc indent for leading asterisk"